integration_tests/src/main/python/date_time_test.py

```diff
@@ -460,8 +460,9 @@ def test_to_timestamp(parser_policy):
         { "spark.sql.legacy.timeParserPolicy": parser_policy})
 
 @pytest.mark.skipif(not is_supported_time_zone(), reason="not all time zones are supported now, refer to https://github.com/NVIDIA/spark-rapids/issues/6839, please update after all time zones are supported")
+# Test years after 1900, refer to issues: https://github.com/NVIDIA/spark-rapids/issues/11543, https://github.com/NVIDIA/spark-rapids/issues/11539
 def test_yyyyMMdd_format_for_legacy_mode():
-    gen = StringGen("[0-9]{3}[1-9](0[1-9]|1[0-2])(0[1-9]|[1-2][0-9])")
+    gen = StringGen('(19[0-9]{2}|[2-9][0-9]{3})([0-9]{4})')
```

Comment:
Q: `(19[0-9]{2}|[2-9][0-9]{3})([0-9]{4})` seems not only modifying the year format, but also changes the `mmdd`?

Reply:
Yes, do not need to change mmdd.

Reply:
you were not replying my question... you changed the `mmdd` in this change.

Reply:
Sorry for the previous reply.
> You changed the mmdd in this change, was this intentional? 

Yes, changed MM, and it's intentional.

The `yyyy` years after 1900 are OK.
For `MM`, invalid values like 99 month will result in NULL. 
For `dd`, also any values `[0-9]{2}` is OK.

This PR is changing branch 24.10, and [PR](https://github.com/NVIDIA/spark-rapids/pull/11544) is changing 24.12.
On branch 24.12, it tests `yyyymmdd` beside of `yyyyMMdd`, and test case name renamed from `test_yyyyMMdd_format_for_legacy_mode` to `test_formats_for_legacy_mode`
mm: Minute.
MM: month.
So we will use the changes on 24.12.
And we expect the auto-merge from 24.10 to 24.12 will fall.